src/__swaps__/screens/Swap/components/GasButton.tsx

```diff
@@ -168,13 +170,20 @@ const GasMenu = ({ children }: { children: ReactNode }) => {
 
 export function ReviewGasButton() {
   const { isDarkMode } = useColorMode();
-  const { SwapNavigation } = useSwapContext();
+  const { SwapNavigation, internalSelectedInputAsset } = useSwapContext();
 
   const separatatorSecondary = useForegroundColor('separatorSecondary');
 
+  const handleShowCustomGas = () => {
+    'worklet';
+
+    runOnJS(setSelectedGasSpeed)(internalSelectedInputAsset.value?.chainId || ChainId.mainnet, 'custom');
```

Comment:
What's the reason of changing this from `runOnUI`  to `runOnJs` ?

Reply:
before it wasn't updating the `setSelectedGasSpeed` here. We have to cross the bridge here either way, I just chose to keep it as a worklet and cross the bridge for updating the selected gas speed since that's kinda hidden while the user interacts with the custom gas sheet.